File: LeetCode/Best_time_to_buy_and_sell_stock_2.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        int Cprofit = 0;
        for(int iter = 0, size = prices.size(); iter < size -1; iter++){
            if(prices[iter] < prices[iter+1]){
                Cprofit = prices[iter+1] - prices[iter];
                if(Cprofit  > 0)
                    profit += Cprofit;
            }
        }
        return profit;
    }
};

File: LeetCode/best_time_to_buy_and_sell_stock_2.java
class Solution {
    public int maxProfit(int[] prices) {
        if(prices.length == 0) return 0;
        int totalProfit = 0;
        int buy_price = prices[0];
        for(int current_price : prices){
            if(current_price > buy_price){
                totalProfit += (current_price - buy_price);
                buy_price = current_price;
            }else{
                buy_price = current_price;
            }
        }
        return totalProfit;
    }
}